### ISSUE TITLE
Generating new accounts with already many accounts

### ISSUE THREAD

**Issue by AdvancedStyle** [paritytech/parity#7230]
> I'm running:
>
> - **Which Parity version?**: 1.8.3-beta
> - **Which operating system?**: Ubuntu 16.04.3 LTS
> - **How installed?**: Release binary
> - **Are you fully synchronized?**: yes
> - **Did you try to restart the node?**: yes

In our parity node there are already 100,000+ account files in the keystore; and we often have need to generate more accounts (many at a time).  For example we generate 100 new accounts using the 
personal_newAccount RPC call.

The problem is that each call seems to trigger an entire keystore directory rescan...which overloads the server.

Better behavior would for parity to only add the new accounts to memory without needing to rescan the keystore.

We have tried resetting the modified time on the keystore directory to avoid rescan (to trick parity into thinking the keystore has not been modified) but it doesn't seem to work.

**Comment by jakecraige** [comment]
We also encountered this issue noticing that the response time of RPC `personal_newAccount` and `personal_listAccounts` calls degrading over time as we used the node and created many accounts.

We were able to locate the source of the problem with both the [`newAccount`](https://github.com/paritytech/parity/blob/master/parity/account.rs#L94) and [`listAccount`](https://github.com/paritytech/parity/blob/master/parity/account.rs#L102) calls initializing a new `EthStore` and calling [`reloadAccounts`](https://github.com/paritytech/parity/blob/master/ethstore/src/ethstore.rs#L300-L306) which iterates over the entire directory. This explains why it slows as we add more accounts.

In our specific case, we don't store the keyfiles with Parity, so we were able to work around the problem by sending a `parity_killAccount` call after we've persisted the keyfile elsewhere.

When we'd like to send a transaction, we'll place the keyfile back on disk, wait for Parity to pick it up using the `personal_listAccounts` and then send the transaction through. This change has made the response times of these calls fairly stable and we don't see it degrade over time anymore.
